### PR TITLE
[fix] Detect `jax.Array` properly

### LINKED ISSUE #2535: Detection of Jax Arrays Breaks on Jax=0.4.*
The detection for jax arrays here https://github.com/aimhubio/aim/blob/285a09fefbf25d418e00bc305e9a1357d6343d3c/aim/sdk/num_utils.py#L66 is broken in Jax 0.4.* since `DeviceArray` was renamed to `Array`. The simple fix to support the prior/current version would be to check for either `DeviceArray` or `Array`

### PR DESCRIPTION
**Pull request by n-gao**

### CODE REVIEW

**Comment by n-gao** [comment]
Done :)

**Review by alberttorosyan — APPROVED** [review]
LGTM

**Comment by alberttorosyan** [comment]
@mihran113 please take a look at this PR. Do you see any potential issues with the type checking?

File: aim/sdk/num_utils.py [hunk 1]
```diff
@@ -67,7 +67,11 @@ def is_jax_device_array(inst):
     """
     Check whether `inst` is instance of jax device array
     """
-    return inst_has_typename(inst, ['jaxlib', 'xla_extension', 'DeviceArray'])
+    if inst_has_typename(inst, ['jaxlib', 'xla_extension', 'Array']):
+        return True
+    if inst_has_typename(inst, ['jaxlib', 'xla_extension', 'DeviceArray']):
+        reutrn True
```
Comment (alberttorosyan) [inline aim/sdk/num_utils.py:73]: @n-gao, a typo here; should be `return`

**Comment by mihran113** [comment]
Hey @n-gao! Thanks a lot for contribution

> @mihran113 please take a look at this PR. Do you see any potential issues with the type checking?

Looks good to me :)

**Review by mihran113 — APPROVED** [review]

**Pull request merged**